eqctl/eqctl.c

```diff
@@ -64,7 +64,13 @@ static int read_setup(unsigned int *data, char setup[], size_t smax)
 	int n_max = smax / sizeof(unsigned int);
 	int separator;
 
+	/* open input file */
 	fh = fopen(setup, "r");
+	if (!fh) {
+		fprintf(stderr, "error: invalid input file name\n");
+		return -EINVAL;
```

Comment:
best return errno and use a more generic error message. See the fopen() man page.